src/dsp_tools/commands/excel2json/properties.py

```diff
@@ -316,9 +337,17 @@ def _do_property_excel_compliance(df: pd.DataFrame, excelfile: str) -> None:
         "gui_element",
         "gui_attributes",
     }
-    utl.check_contains_required_columns_else_raise_error(df=df, required_columns=required_columns)
-    utl.check_column_for_duplicate_else_raise_error(df=df, to_check_column="name")
-    _check_missing_values_in_row_raise_error(df=df, excelfile=excelfile)
+    problems = [
+        utl.check_contains_required_columns_else_raise_error(df=df, required_columns=required_columns),
+        utl.check_column_for_duplicate(df=df, to_check_column="name"),
+    ]
+    if missing_vals_check := _check_missing_values_in_row(df=df):
+        problems.extend(missing_vals_check)
+    if any(problems):
+        extra = [problem.execute_error_protocol() for problem in problems if problem]
+
+        msg = [f"The excel file '{excelfile}' has some problems:", *extra]
```

Comment:
My thinking was along these lines:
- we have a `RequiredColumnMissingProblem` which can be checked and generated by one function
- we have a `DuplicateInColumnProblem`  which also gets created by a function
- we have a list of  `MissingRowValuesProblem` which gets created by a function
- we have an over-arching `ExcelPropertyProblem` (or whatever a good name might be) wrapper which contains the three above
- we have a factory function that takes the three optional inner problems and returns optionally a wrapper problem; the semantics would be: if none of the inner problems, then the wrapper would be none; if any problem, then the wrapper would be created containing at least one problem.
- the wrapper would provide you with an `execute_error_protocol()` that delegates this call to all of its members

Does that make some kind of sense?

Reply:
no, I didn't mean in terms of subclasses, but in terms of a dataclass representing the aggregated problem, that has three members, which each are dataclasses representing the sub-problems

Reply:
I decided to leave out an aggregator class. The only further information it would have is the excel name. As the object is not currently handed around there seemed to me too little benefit to create an additional class, so I decided to add the name directly in the error msg string.

Reply:
for now, fair enough. In the long run we may want to re-think the fact that this method raises the error rather than just returning the problem directly